itou/www/invitations_views/tests/tests_siae_accept.py

```diff
@@ -82,6 +82,48 @@ def test_accept_invitation_signup_(self):
         user = User.objects.get(email=invitation.email)
         self.assert_accepted_invitation(response, invitation, user)
 
+    @respx.mock
+    def test_accept_invitation_signup_bad_email_case(self):
+        invitation = SentSiaeStaffInvitationFactory(email=OIDC_USERINFO["email"].upper())
+        response = self.client.get(invitation.acceptance_link, follow=True)
+        self.assertContains(response, "logo-inclusion-connect-one-line.svg")
+
+        # We don't put the full path with the FQDN in the parameters
+        previous_url = invitation.acceptance_link.split(settings.ITOU_FQDN)[1]
+        next_url = reverse("invitations_views:join_siae", args=(invitation.pk,))
+        params = {
+            "user_kind": KIND_SIAE_STAFF,
+            "user_email": invitation.email,
+            "channel": "invitation",
+            "previous_url": previous_url,
+            "next_url": next_url,
+        }
+        url = escape(f"{reverse('inclusion_connect:authorize')}?{urlencode(params)}")
+        self.assertContains(response, url + '"')
+
+        total_users_before = User.objects.count()
+
+        response = mock_oauth_dance(
+            self,
+            KIND_SIAE_STAFF,
+            assert_redirects=False,
+            # Using the same email with a different case should not fail
+            user_email=invitation.email.lower(),
+            channel="invitation",
+            previous_url=previous_url,
+            next_url=next_url,
+        )
+        response = self.client.get(response.url, follow=True)
+        # Check user is redirected to the welcoming tour
+        last_url, _ = response.redirect_chain[-1]
+        assert last_url == reverse("welcoming_tour:index")
+
+        total_users_after = User.objects.count()
+        assert (total_users_before + 1) == total_users_after
+
+        user = User.objects.get(email=invitation.email)
```

Comment:
J'ai l'impression qu'ici on ne veux tester que l'utilisateur et pas le nombre d'utilisateurs donc peut-être remplacer le premier `User.objects.count()` par `User.objects.filter(...).first() is None` (ou `.count() == 0` :grinning:) et faire disparaître le second au profit du `User.objects.get(...)`.